coreax/util.py

```diff
@@ -97,7 +122,10 @@ def squared_distance(x: ArrayLike, y: ArrayLike) -> Array:
     return jnp.dot(x - y, x - y)
 
 
-@jit
+@deprecated(
+    "Use coreax.util.pairwise(coreax.util.squared_distance)(x, y);"
+    "will be removed in version 0.3.0"
```

Comment:
We could keep the old functions if needed. However, to keep things consistent/simpler across the codebase, I thought it would be better to deprecate the old specialised calls (this pull removes all usage of them). Regarding the specifics, two minor versions ahead is the same deprecation window we have used before. We could delete these entirely, but as they are easy to provide proper deprecation for, I though this would be the best way.